lib/score_card.js

```diff
@@ -0,0 +1,119 @@
+function ScoreCard(){
+  this.total = 0
+  this.frame = new Frame()
+  this.frames = []
+  this.frameCount = 1
+}
+
+ScoreCard.prototype.recordRoll = function(points){
+  this.frameCheck()
+  if(this.cheatGuard(points)==true){
+    return 'illegal roll'
+  } else {
+    if(this.frameCount > 10){
+      this.addBonus(points)
+    } else {
+      this.addPoints(points)
+}
+}
+}
+
+
+ScoreCard.prototype.addBonus = function(points){
```

Comment:
It feels like there's a lot of if/else here, in the next method and elsewhere in the code. A lot of what you're doing in the scorecard is manipulating the frame based on different game states. In Ruby, you may have come across the principle 'Tell, don't ask' (see https://robots.thoughtbot.com/tell-dont-ask) - the same principle applies. 

Rather than 'If we have a spare bonus, then, if we have a strike bonus and it's the last frame, then' etc, you should be aiming for 'Hey frame, here's the score - deal with it'. 

You can achieve that by better encapsulating this logic within the frame - so you end up just calling frame.roll(numberOfPinsKnockedDown) and letting frame dealing with the rest.

Of course, you still need to deal with bonuses, and one frame doesn't need to know that another frame is waiting for a bonus. This is where we need to create game rules (as we discussed in code review on Monday), we we could have a StrikeBonusRule and a SpareBonusRule and our code becomes:

```
frame.roll(numberOfPinsToKnockDown)
rules.forEach((rule) => {
  rules.apply(frames, numberOfPinsKnockedDown)
});
```
And an example rule might be
```
function StrikeBonusRule() {
   function apply(frames, numberOfPinsKnockedDown) {
      var framesAwaitingBonus = frames.filter((frame) => {
         return frame.needsStrikeBonus();
      })
      framesAwaitingBonus.addStrikeBonus(numberOfPinsKnockedDown);
   }
   return {
      apply: apply
   }
}
```